inference-engine/src/vpu/graph_transformer/src/stages/interpolate.cpp

```diff
@@ -0,0 +1,85 @@
+    if (input->desc().dimsOrder() == DimsOrder::NCHW || input->desc().dimsOrder() == DimsOrder::NHWC ||
+        input->desc().dimsOrder() == DimsOrder::CHW || input->desc().dimsOrder() == DimsOrder::HWC) {
+        const auto ic = input->desc().dim(Dim::C);
+        const auto in = (input->desc().numDims() == 3) ? 1 : input->desc().dim(Dim::N);
+
+        const auto oc = output->desc().dim(Dim::C);
+        const auto on = (output->desc().numDims() == 3) ? 1 : output->desc().dim(Dim::N);
+
+        auto padsBegin = _layer->GetParamAsInts("pads_begin", {});
+        auto padsEnd = _layer->GetParamAsInts("pads_end", {});
+
+        const auto isPadZeros = [](const std::vector<int>& pad) {
+            return std::all_of(pad.begin(), pad.end(), [](int i) { return i == 0; });
+        };
+
+        if (ic == oc && in == 1 && on == 1 && isPadZeros(padsBegin) && isPadZeros(padsEnd)) {
+            ie::details::CaselessEq<std::string> cmp;
+            if (cmp(interpolateMode, "nearest")) {
+                // current "Resample" supports the following "Interpolate" modes only:
+                // coordinate_transformation_mode = half_pixel; nearest_mode = round_prefer_ceil;
+                // other "Interpolate" modes are translated to the default ones
+                const auto antialias = _layer->GetParamAsBool("antialias", false);
+                _stageBuilder->addResampleNearestStage(model,
+                                                       _layer->name,
+                                                       _layer,
+                                                       antialias,
+                                                       -1.0f,
+                                                       input,
+                                                       output);
+            } else if (cmp(interpolateMode, "linear")) {
+                // current "Interp" supports modes "align_corners" and "asymetric" only
+                const auto coordinate_transformation_mode = _layer->GetParamAsString("coordinate_transformation_mode", "half_pixel");
+
+                _stageBuilder->addInterpStage(model,
+                                              _layer->name,
+                                              _layer,
+                                              cmp(coordinate_transformation_mode, "align_corners"),
+                                              input,
+                                              output);
+            } else {
+                VPU_THROW_EXCEPTION << "Current Interpolate supports 'nearest' and 'linear' modes only; layer name = " << _layer->name;
+            }
+        } else {
+            VPU_THROW_EXCEPTION << "Current Interpolate does not support paddings, batches, and resize by channels; layer name = " << _layer->name;
+        }
+    } else {
+        VPU_THROW_EXCEPTION << "Current Interpolate supports (N)HWC, (N)CHW data orders only; layer name = " << _layer->name;
+    }
```

Comment:
Nitpick: you can use `VPU_THROW_UNLESS(input->desc().dimsOrder() == DimsOrder::NCHW || input->desc().dimsOrder() == DimsOrder::NHWC ||
        input->desc().dimsOrder() == DimsOrder::CHW || input->desc().dimsOrder() == DimsOrder::HWC, "blah")` (the same with other conditions) and there would be not need to do three-story nested conditions

Reply:
I would prefer to leave the current assert. This one is quite huge, and error message is not split.